// @counterfactual/types/dist/node-protocol.js
export var Node;
(function (Node) {
    let ErrorType;
    (function (ErrorType) {
        ErrorType["ERROR"] = "error";
    })(ErrorType = Node.ErrorType || (Node.ErrorType = {}));
    let MethodName;
    (function (MethodName) {
        MethodName["ACCEPT_STATE"] = "acceptState";
        MethodName["CREATE_CHANNEL"] = "createChannel";
        MethodName["DEPOSIT"] = "deposit";
        MethodName["GET_APP_INSTANCE_DETAILS"] = "getAppInstanceDetails";
        MethodName["GET_APP_INSTANCES"] = "getAppInstances";
        MethodName["GET_CHANNEL_ADDRESSES"] = "getChannelAddresses";
        MethodName["GET_FREE_BALANCE_STATE"] = "getFreeBalanceState";
        MethodName["GET_MY_FREE_BALANCE_FOR_STATE"] = "getMyFreeBalanceForState";
        MethodName["GET_PROPOSED_APP_INSTANCES"] = "getProposedAppInstances";
        MethodName["GET_STATE"] = "getState";
        MethodName["INSTALL"] = "install";
        MethodName["INSTALL_VIRTUAL"] = "installVirtual";
        MethodName["PROPOSE_INSTALL"] = "proposeInstall";
        MethodName["PROPOSE_INSTALL_VIRTUAL"] = "proposeInstallVirtual";
        MethodName["PROPOSE_STATE"] = "proposeState";
        MethodName["REJECT_INSTALL"] = "rejectInstall";
        MethodName["REJECT_STATE"] = "rejectState";
        MethodName["UPDATE_STATE"] = "updateState";
        MethodName["TAKE_ACTION"] = "takeAction";
        MethodName["UNINSTALL"] = "uninstall";
        MethodName["UNINSTALL_VIRTUAL"] = "uninstallVirtual";
        MethodName["WITHDRAW"] = "withdraw";
    })(MethodName = Node.MethodName || (Node.MethodName = {}));
    let EventName;
    (function (EventName) {
        EventName["COUNTER_DEPOSIT_CONFIRMED"] = "counterDepositConfirmed";
        EventName["CREATE_CHANNEL"] = "createChannelEvent";
        EventName["DEPOSIT_CONFIRMED"] = "depositConfirmedEvent";
        EventName["DEPOSIT_FAILED"] = "depositFailed";
        EventName["DEPOSIT_STARTED"] = "depositStartedEvent";
        EventName["INSTALL"] = "installEvent";
        EventName["INSTALL_VIRTUAL"] = "installVirtualEvent";
        EventName["PROPOSE_STATE"] = "proposeStateEvent";
        EventName["REJECT_INSTALL"] = "rejectInstallEvent";
        EventName["REJECT_STATE"] = "rejectStateEvent";
        EventName["UNINSTALL"] = "uninstallEvent";
        EventName["UNINSTALL_VIRTUAL"] = "uninstallVirtualEvent";
        EventName["UPDATE_STATE"] = "updateStateEvent";
        EventName["WITHDRAWAL_CONFIRMED"] = "withdrawalConfirmedEvent";
        EventName["WITHDRAWAL_FAILED"] = "withdrawalFailed";
        EventName["WITHDRAWAL_STARTED"] = "withdrawalStartedEvent";
        EventName["PROPOSE_INSTALL"] = "proposeInstallEvent";
        EventName["PROPOSE_INSTALL_VIRTUAL"] = "proposeInstallVirtualEvent";
        EventName["PROTOCOL_MESSAGE_EVENT"] = "protocolMessageEvent";
        EventName["WITHDRAW_EVENT"] = "withdrawEvent";
        EventName["REJECT_INSTALL_VIRTUAL"] = "rejectInstallVirtualEvent";
    })(EventName = Node.EventName || (Node.EventName = {}));
})(Node || (Node = {}));
